src/components/picker/index.tsx

```diff
@@ -93,6 +96,19 @@ const Picker = React.forwardRef((props: PickerProps, ref) => {
   } = themeProps;
   const {preset} = others;
 
+  const {
+    type,
+    headerProps,
+    renderCustomModal: typeRenderCustomModal,
+    dialogProps,
+    pickerModalProps
+    //@ts-expect-error
+  } = usePickerMode({
+    pickerType: 'modal',
```

Comment:
If you're passing a default here it will never go to the else in the usePickerMode...

Reply:
yes I know, for know we won't set it as `default prop` since typing issues ( I spoke with Ethan about it).
passing it to the hook will return the right type.

Reply:
Not sure I follow... I see you fixed it by checking in the hook if the pickerType prop was passed else you determine the type using deprecated props and have it defaults to 'modal' - that's good

Reply:
mark as resolved